servicetalk-opentelemetry-http/src/main/java/io/servicetalk/opentelemetry/http/RequestHeadersPropagatorSetter.java

```diff
@@ -31,7 +33,12 @@ private RequestHeadersPropagatorSetter() {
     @Override
     public void set(@Nullable final RequestInfo requestInfo, final String key, final String value) {
         if (requestInfo != null) {
-            requestInfo.request().headers().set(key.toLowerCase(Locale.ENGLISH), value);
+            set(requestInfo.request().headers(), key, value);
```

Comment:
Here, as we have full `RequestInfo`, consider first checking `requestInfo.connectionInfo().protocol()` (if it's `HttpProtocolVersion`), then fallback to `requestInfo.request().version()`. Do `toLowerCase` only if protocol's major is >= 0.